### PR TITLE
Adding vehicle classes

### PR DESCRIPTION
**Pull request by Ronenii**
Finished Adding the basic logic of all the vehicle classes.
TODO:
- Write Value out of range exception
- Write Garage.cs logic
- Add comments where needed
- Handle exceptions properly (right now methods only throw exceptions, I didn't add any output or catching)
- Combine UI and Logic
- Change garage's name to "Microsoft John's Garage"

### CODE REVIEW

File: Ex03.GarageLogic/Wheel.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+
+namespace Ex03.GarageLogic
+{
+    class Wheel
+    {
+        private readonly float r_MaxPSI;
+        private readonly string r_Manufacturer;
```
Comment (idobi111) [inline Ex03.GarageLogic/Wheel.cs:12]: why this is readonly?

File: Ex03.GarageLogic/Vehicle.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+{
+    public enum eEngineType
+    {
+        Diesel,
+        Electric
+    }
+    public abstract class Vehicle
+    {
+        private readonly string r_Model;
+        private readonly string r_LicensePlate;
```
Comment (idobi111) [inline Ex03.GarageLogic/Vehicle.cs:42]: why readonly?

File: Ex03.GarageLogic/Vehicle.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        private float m_EnergyLeft;
+
+        private List<Wheel> m_Wheels;
+
+        public Vehicle(string i_Model, string i_LicensePlate, float i_EnergyLeft ,float i_MaxPSI, string i_WheelManufacturer,float i_CurrentPsi , int i_NumOfWheels)
+        {
+            r_Model = i_Model;
+            r_LicensePlate = i_LicensePlate;
+            m_EnergyLeft = i_EnergyLeft;
+            createAllWheels(i_NumOfWheels, i_CurrentPsi, i_WheelManufacturer, i_MaxPSI);
```
Comment (idobi111) [inline Ex03.GarageLogic/Vehicle.cs]: make it simple for now

we does not need to create all wheels in the ctor
meanwhile in my opinion we should get list of wheels to the ctor and make 
m_wheels = i_wheels

File: Ex03.GarageLogic/Truck.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Ex03.GarageLogic
+{
+    class Truck : DieselVehicle
```
Comment (idobi111) [inline Ex03.GarageLogic/Truck.cs]: add public

File: Ex03.GarageLogic/Truck.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        private const int k_NumOfWheels = 14;
+        private const float k_MaxPSI = 26f;
+        private const eFuelType k_FuelType = eFuelType.Soler;
+        private const float k_MaxFuelLiters = 135f;
+
+        private readonly bool r_IsTransportingHazardousMaterial;
+        private readonly float r_CargoSize;
```
Comment (idobi111) [inline Ex03.GarageLogic/Truck.cs:16]: why readonly?

File: Ex03.GarageLogic/ElectricCar.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+using System.Linq;
+using System.Text;
+
+namespace Ex03.GarageLogic
+{
+    class ElectricCar: ElectricVehicle
+    {
+        private const int k_NumOfWheels = 5;
+        private const float k_MaxPSI = 33f;
+        private const float k_MaxChargeHours = 2.6f;
```
Comment (idobi111) [inline Ex03.GarageLogic/ElectricCar.cs]: should be 5.2

File: Ex03.GarageLogic/DieselCar.cs [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace Ex03.GarageLogic
+{
+    public enum eCarColor
+    {
+        White,
+        Black,
+        Yellow,
+        Red
+    }
```
Comment (idobi111) [inline Ex03.GarageLogic/DieselCar.cs]: should locate in vehicle

File: Ex03.GarageLogic/DieselCar.cs [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    public enum eNumOfCarDoors
+    {
+        Two =2,
+        Three,
+        Four,
+        Five
+    }
```
Comment (idobi111) [inline Ex03.GarageLogic/DieselCar.cs]: should locate in the vehicle

File: Ex03.GarageLogic/DieselBike.cs [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace Ex03.GarageLogic
+{
+    public enum eBikeLicense
+    {
+        A1,
+        A2,
+        AA,
+        B1
+    }
```
Comment (idobi111) [inline Ex03.GarageLogic/DieselBike.cs]: should locate in vehicle

File: Ex03.GarageLogic/Vehicle.cs (hunk 3 above)
Comment (Ronenii) [reply Ex03.GarageLogic/Vehicle.cs]: I did this but we now have to figure out how to use the given wheel properties in the ctor. Or if we need them at all when we create a new vehicle

**Comment by Ronenii** [comment]
Made the required changes.
I added "Car" and "Bike" as classes so we could use a constructor. They can also be a struct I just did this for comfort.

**Review by idobi111 — APPROVED** [review]

**Pull request merged**